src/interchange/toml/lexer.zig

```diff
@@ -586,41 +584,15 @@ pub const Lexer = struct {
                     lexer.token = T.t_comma;
                 },
                 ';' => {
-                    if (lexer.has_newline_before) {
-                        lexer.step();
-
-                        singleLineComment: while (true) {
-                            lexer.step();
-                            switch (lexer.code_point) {
-                                '\r', '\n', 0x2028, 0x2029 => {
-                                    break :singleLineComment;
-                                },
-                                -1 => {
-                                    break :singleLineComment;
-                                },
-                                else => {},
-                            }
-                        }
-                        continue;
+                    if (!lexer.has_newline_before) {
+                        try lexer.addDefaultError("Unexpected semicolon");
                     }
 
-                    try lexer.addDefaultError("Unexpected semicolon");
+                    lexer.discardTilEndOfLine();
+                    continue;
```

Comment:
These don't seem equivalent to me.

I think the original implementation was:

```
if newline-before:
  flush();
else:
  ...
fail();
```

The new implementation seems to be:
```
if newline-before:
  ...
else:
  fail();
flush();
```

Reply:
Though I haven't written a ton of golang in my life, I picked this (keep the happy path left-aligned) up from that community and find it quite nice to read most of the time.